server/routes/playlist.js

```diff
@@ -195,4 +191,58 @@ router.post('/removeSimilarityCount', jwt.verifyJWT, async (req, res, next) => {
   }
 })
 
+/* calculates and saves similarity scores between playlist and
+all other playlists in database */
+router.post('/save-similarity-score', jwt.verifyJWT, async (req, res, next) => {
+  try {
+    const userId = req.userId
+    const { playlistId, trackVector } = req.body
+
+    const allPlaylists = await Playlist.find({ _id: { $ne: userId }, added: true })
+
+    for (let i = 0; i < allPlaylists.length; i++) {
```

Comment:
Try using map

Reply:
I don't think I can use map because I want to await before going on to the next iteration, and I think I can only do that with a for loop.